### PR TITLE
Add figure object to return statement of `draw` method of `EGraph`

### PR DESCRIPTION
**Pull request by soosub**
## Context for changes

Analogous to #33, it is useful to return the figure object to the user. This PR adds the figure object to return statement of `draw` method of `EGraph`.

- **Improvements**:
    * Figure object included in return statement of `EGraph.draw`
- **Documentation changes**:
    - [x] added documentation for the relevant methods

## Example usage and tests

- [x] Not Applicable

## Performance results justifying changes

- [x] Not Applicable

## Workflow actions and tests

- [x] To be written

## Expected benefits and drawbacks

**Expected benefits:**
- Allows user more flexibility with using the figure and saving it for example.

**Possible drawbacks:**
- Unforeseen changes where the return of the current implementation is used. If I missed calls these can easily be fixed by simply using the second index of the new return tuple.

## Related Github issues
#46 

## Checklist and integration statements

- [x] My Python and C++ codes follow the coding and commenting styles of this project as indicated by existing files. Specifically, the changes conform to given `black`, `docformatter` and `pylint` configurations. 
- [x] I have performed a self-review of these changes, checked my code (including for [codefactor](https://www.codefactor.io/repository/github/xanaduai/flamingpy/branches) compliance), and corrected misspellings to the best of my capacity. I have synced this branch with others as required.
- [x] I have added context for corresponding changes in documentation and [`README.md`](README.md) as needed.
- [x] I have added new workflow CI tests for corresponding changes, ensuring codecoverage 95% or better, and these pass locally for me.
- [x] I have updated [`CHANGELOG.md`](.github/CHANGELOG.md) following the template. I recognize that the developers may revisit [`CHANGELOG.md`](.github/CHANGELOG.md) and the versioning, and create a Special Release including my changes.

### CODE REVIEW

**Comment by soosub** [comment]
@ilan-tz during the writing of `test_draw` I noticed that the user is allowed to make an EGraph where the nodes are not given as (x,y,z) coordinates. Should we continue allowing this or force the user specify the nodes as a 3-tuple in the EGraph? This is mainly a problem for the corresponding draw method

**Comment by ilan-tz** [comment]
> @ilan-tz during the writing of `test_draw` I noticed that the user is allowed to make an EGraph where the nodes are not given as (x,y,z) coordinates. Should we continue allowing this or force the user specify the nodes as a 3-tuple in the EGraph? This is mainly a problem for the corresponding draw method

The EGraph can work with any kind of hashable node since it's a NetworkX graph. However, most methods and most FP functionality does not make sense for non-tuple nodes, so I think it makes sense for us to enforce 3-tuples.

**Comment by soosub** [comment]
> > @ilan-tz during the writing of `test_draw` I noticed that the user is allowed to make an EGraph where the nodes are not given as (x,y,z) coordinates. Should we continue allowing this or force the user specify the nodes as a 3-tuple in the EGraph? This is mainly a problem for the corresponding draw method
> 
> The EGraph can work with any kind of hashable node since it's a NetworkX graph. However, most methods and most FP functionality does not make sense for non-tuple nodes, so I think it makes sense for us to enforce 3-tuples.

Alright, then I'll make an issue out of it 👍🏼

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ See full commit details ...
 * Our frontend simulator script, [`simulations.py`](flamingpy/simulations.py), now supports simple and highly-scalable MPI jobs through `mpi4py` libraries in a non-intrusive manner. The users who do **not** have or want MPI, can run `simulations.py` single-threaded as per usual without facing any errors. MPI users can speed up Monte Carlo samplings in EC steps virtually up to as many processors they can throw at it. The script support jobs both on local machines and large-scale clusters. [#47](https://github.com/XanaduAI/flamingpy/pull/47) (backward compatible)
   * MPI users on their local machines can simply run the following for a 4-processor job:
   `mpirun -np 4 python flamingpy/simulations.py`
+* Added `fig, ax` returns for the draw methods in `utils/viz.py`.
```
Comment (ilan-tz) [inline .github/CHANGELOG.md:37]:
```suggestion
* Added `fig, ax` returns for the draw methods in `utils/viz.py`.  [#55](https://github.com/XanaduAI/flamingpy/pull/55)
```

File: tests/unit/test_graphstates.py [hunk 2]
```diff
@@ -82,8 +83,13 @@ def test_index(self, random_graph):
     # def test_slice_coords(self):
     # pass
 
-    # def test_draw(self):
-    # pass
+    def test_draw(self):
```
Comment (ilan-tz) [inline tests/unit/test_graphstates.py:86]: In reality this is testing the draw_EGraph in the viz module, so we should move this tests to the viz test files.

**Review by ilan-tz — CHANGES_REQUESTED** [review]
Nice! Only very minor comments.

File: tests/unit/test_viz.py [hunk 3]
```diff
@@ -53,35 +53,45 @@ def test_to_pi_string():
...
 
 
-def test_draw_egraph_bell():
-    """Test for the draw method of EGraph of Bell state."""
-    # Bell state EGraph
-    edge = [(0, 0, 0), (0, 0, 1)]
-    bell_state = EGraph()
-    bell_state.add_edge(*edge, color="MidnightBlue")
+class TestDrawEGraph:
+    """Tests for EGraphs."""
```
Comment (ilan-tz) [inline tests/unit/test_viz.py:57]:
```suggestion
    """Tests for visualizing EGraphs."""
```

**Review by ilan-tz — APPROVED** [review]

**Pull request merged**